x-pack/plugins/apm/typings/es_schemas/ui/fields/agent.ts

```diff
@@ -29,6 +27,12 @@ export type AgentName =
   | 'opentelemetry/ruby'
   | 'opentelemetry/webjs';
 
+/*
+ * Support additional agent types by appending definitions in mappings.json
+ * (for telemetry) and the AgentName type.
+ */
+export type AgentName = ElasticAgentName | OpenTelemetryAgentName;
```

Comment:
@afharo if the telemetry tools support aliases, we might have a bug. I worked around the `Unknown type TypeParameter; T` error thrown when extracting the updated schema by explicitly typing `AgentName` with the combined options from `ElasticAgentName` and `OpenTelemetryAgentName`.